File: GeneratedEntityReaderClasses/Components/GameStatsComponent.cs
namespace GeneratedNoitaClasses;
public class GameStatsComponent : NoitaComponentBase,  iNoitaType<GameStatsComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString name { get; set; } = new();
	public NoitaString stats_filename { get; set; } = new();
	public NoitaBool is_player { get; set; } = new();
	public NoitaString extra_death_msg { get; set; } = new();
	public NoitaBool dont_do_logplayerkill { get; set; } = new();
	public NoitaInt player_polymorph_count { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		name.Read(s);
		stats_filename.Read(s);
		is_player.Read(s);
		extra_death_msg.Read(s);
		dont_do_logplayerkill.Read(s);
		player_polymorph_count.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		name.Write(s);
		stats_filename.Write(s);
		is_player.Write(s);
		extra_death_msg.Write(s);
		dont_do_logplayerkill.Write(s);
		player_polymorph_count.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/GameStatsComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class GameStatsComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    name:NoitaString = field(default_factory=NoitaString)
    stats_filename:NoitaString = field(default_factory=NoitaString)
    is_player:NoitaBool = field(default_factory=NoitaBool)
    extra_death_msg:NoitaString = field(default_factory=NoitaString)
    dont_do_logplayerkill:NoitaBool = field(default_factory=NoitaBool)
    player_polymorph_count:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.name.Read(s)
        self.stats_filename.Read(s)
        self.is_player.Read(s)
        self.extra_death_msg.Read(s)
        self.dont_do_logplayerkill.Read(s)
        self.player_polymorph_count.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.name.Write(s)
        self.stats_filename.Write(s)
        self.is_player.Write(s)
        self.extra_death_msg.Write(s)
        self.dont_do_logplayerkill.Write(s)
        self.player_polymorph_count.Write(s)

